k2/csrc/fsa_algo.cc

```diff
@@ -147,4 +147,109 @@ void Connect(const Fsa &a, Fsa *b, std::vector<int32_t> *arc_map /*=nullptr*/) {
   }
 }
 
+bool TopoSort(const Fsa &a, Fsa *b,
+              std::vector<int32_t> *state_map /*= nullptr*/) {
+  CHECK_NOTNULL(b);
+  b->arc_indexes.clear();
+  b->arcs.clear();
+
+  if (state_map) state_map->clear();
```

Comment:
Actually I prefer `state_map != nullptr` but I did't find the rule in Google Style, just an example in

https://google.github.io/styleguide/cppguide.html

Reply:
I assume you are referring to https://google.github.io/styleguide/cppguide.html#Run-Time_Type_Information__RTTI_

Tensorflow also uses
```cpp
if (x != nullptr)
```
Please see
- https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.cc#L177
- https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.cc#L213
- https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.cc#L251

-----

I will change it to `!= nullptr`

Reply:
Tensorflow also uses

- https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.cc#L696
```cpp
  if (in_size == 0) {
```

----

Maybe we can also follow this style.

```cpp
if (x == 0)

if (b == false)
```

Reply:
Yes, I like all of those above except `if (bool_value == false)`, I prefer `if (!bool_value)` for boolean values, as `if` is checking true or false and boolean value is already a `true` or `false`.

Another example is that in STL code there is a lot of `empty()` function and people tend to call `if(v.empty())`

Reply:
Done.